builtin/logical/database/path_roles.go

```diff
@@ -355,20 +322,65 @@ func (b *databaseBackend) pathRoleCreateUpdate(ctx context.Context, req *logical
 		role = &roleEntry{}
 	}
 
-	if err := pathRoleCreateUpdateCommon(ctx, role, req.Operation, data); err != nil {
-		return logical.ErrorResponse(err.Error()), nil
+	createOperation := (req.Operation == logical.CreateOperation)
+
+	// DB Attributes
+	{
+		if dbNameRaw, ok := data.GetOk("db_name"); ok {
+			role.DBName = dbNameRaw.(string)
+		} else if createOperation {
+			role.DBName = data.Get("db_name").(string)
+		}
+		if role.DBName == "" {
+			return logical.ErrorResponse("database name is required"), nil
+		}
+	}
+
+	// Statements
+	{
+		if creationStmtsRaw, ok := data.GetOk("creation_statements"); ok {
+			role.Statements.Creation = creationStmtsRaw.([]string)
+		} else if createOperation {
+			role.Statements.Creation = data.Get("creation_statements").([]string)
+		}
```

Comment:
Are there any databases plugins that don't require a create statement? Should we verify here that it exists?

Reply:
Resolved this because I figured out it's not required.

Reply:
This change here is a revert to how creating normal dynamic roles was before merging the static account feature. Before this PR, static roles supported `creation_statements` but now they don't. 

I'm not 100% sure I know the answer to your question, but I do think it's outside the scope of this PR, as it's going back to the way things were pre-https://github.com/hashicorp/vault/pull/6834